### PR TITLE
Describe context definitions and hide "hidden" attribute of contexts

### PR DESCRIPTION
**Pull request by TrangEmily**

### CODE REVIEW

File: adr/context.py [hunk 1]
```diff
@@ -5,6 +5,26 @@
...
+#           'help': help_text
+#          }]
+# ]
+# The display order of contexts on webapp follows the order of contexts in COMMON_CONTEXTS
+# If action is append, this context on webapp will be multiple-choices
+# Define context in query/recipe:
+# 1. If define a new context (not in COMMON_CONTEXTS), follow full definition template
+# 2. If override some attribute of an existing context, use override function
+# Attribute "hidden": if a context of a recipe has "hidden: True",
+#                     it will not be listed on webapp and cli
```
Comment (ahal) [inline adr/context.py]: Might as well turn this into a module-level docstring.

File: adr/query.py [hunk 2]
```diff
@@ -34,7 +34,10 @@ def __init__(self, definitions):
             elif len(definition) >= 2:
                 # Set destination from name
                 definition[1]['dest'] = name
-                self.add_argument(*definition[0], **definition[1])
+                if ("hidden" in definition[1]) and (definition[1]['hidden'] is True):
+                    self.add_argument(*definition[0], help=SUPPRESS)
```
Comment (ahal) [inline adr/query.py]: I think you'll still need to pass all the other values of `definition[1]` into `add_argument`. I think you need something like this:

    if definition[1].get('hidden'):
        del definition[1]['hidden']
        definition[1]['help'] = SUPPRESS

    self.add_argument(*definition[0]. **definition[1])

File: adr/recipes/skipped_tests.py [hunk 3]
```diff
@@ -8,19 +8,19 @@
...
 
+from ..context import override
 from ..query import run_query
 
+# RUN_CONTEXTS = {'limit': {'default': 25,
+#                           'help': "Maximum number of users in result"},
+#                 'sort_key': {'default': 1,
+#                              'help': "Key to sort on (int, 0-based index)",
+#                              }
+#                 }
```
Comment (ahal) [inline adr/recipes/skipped_tests.py]: Remove this commented out block.

File: adr/context.py (hunk 1 above)
Comment (TrangEmily) [reply adr/context.py]: Thanks, I updated, but not sure whether it is what you suggested or not.

File: adr/query.py (hunk 2 above)
Comment (TrangEmily) [reply adr/query.py]: Yes, this is a bug. The hidden attribute still need default value. I updated it.

File: adr/recipes/skipped_tests.py (hunk 3 above)
Comment (TrangEmily) [reply adr/recipes/skipped_tests.py]: I removed in the second commit yesterday, but thank anyway, I often forgot to delete commented code :)

**Pull request merged**